Zagrios/bs-manager
# Get running processes independent of platform

## Scope

Currently, running processes are determined by Windows registry keys, and by executing the `tasklist` command. This means that none of those functions work under Linux. This PR aims to fix that issue.

## Implementation

I've brought in the [ps-list](https://www.npmjs.com/package/ps-list) package, which is a platform-independent way of getting all running processes on the system. 

`SteamService#getActiveUser` has been removed, and `SteamService#steamRunning` now uses `ps-list` to check for any processes with `steam` in the name.

`UtilsService#taskRunning` has also been refactored to work similarly. With this refactor, it also had to be made async, bringing a few other methods along for the ride.

## How to Test

I'm not able to test this change on Windows, only Linux. Things appear to be functioning in my testing, but further testing should be done.

The easiest way to test these changes is to perform the following actions:

- Launch an instance with Steam running.
- Launch an instance with Steam closed.
- Launch an instance with another instance running.

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |
